library/blob.c

```diff
@@ -1625,22 +1646,16 @@ static int eblob_write_prepare_disk_ll(struct eblob_backend *b, struct eblob_key
 		goto err_out_rollback;
 
 	/*
-	 * If no footer is set then commit phase would be skipped and
-	 * so iterator could consider record broken because offset+size
-	 * may be outside of blob. So extend blob manually.
-	 *
-	 * Also we need to extend blob if copy if it was requested, because it
-	 * may be plain_write call that does not call commit and thus following
-	 * copy may try to access area outside of base.
+	 * Allocates space for the record. It should be done because if commit phase will be skipped
+	 * or eblob will be restarted before commit phase, the iterator will consider the record broken
+	 * because offset + size may be outside of blob. So extend blob manually.
```

Comment:
The second part of the comment (albeit in bad english) is still valid, isn't it?